examples/connext_dds/asynchronous_publication/py/async_publisher.py

```diff
@@ -0,0 +1,134 @@
+#
+# (c) 2022 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+#
+# RTI grants Licensee a license to use, modify, compile, and create derivative
+# works of the Software solely for use with RTI products.  The Software is
+# provided "as is", with no warranty of any type, including any warranty for
+# fitness for any purpose. RTI is under no obligation to maintain or support
+# the Software.  RTI shall not be liable for any incidental or consequential
+# damages arising out of the use or inability to use the software.
+#
+import argparse
+import time
+
+import rti.connextdds as dds
+
+from async_type import async_type
+
+
+def run_publisher_application(
+    domain_id: int, sample_count: int, use_xml_qos: bool = False
+):
+    # Start communicating in a domain, usually one participant per application
+    participant = dds.DomainParticipant(domain_id)
+
+    # Create a Topic
+    topic = dds.Topic(participant, "Example async", async_type)
+
+    if use_xml_qos:
+        # Retrieve the default DataWriter QoS, from USER_QOS_PROFILES.xml
+        writer_qos = dds.QosProvider.default.datawriter_qos
+    else:
+        # Configure the DataWriterQos in code, to the same effect as the XML file.
+        writer_qos = participant.default_datawriter_qos
+        writer_qos.reliability = dds.Reliability.reliable(
+            dds.Duration.from_seconds(60)
+        )
+        # Note: Admin Console is reporting history depth of 1 still regardless of changes in the QoS file or programatically
+        # writer_qos.history. = dds.History.keep_last(12)
+        writer_qos.history.kind = dds.HistoryKind.KEEP_LAST
+        writer_qos.history.depth = 12
+        writer_qos.data_writer_protocol.rtps_reliable_writer.min_send_window_size = (
+            50
+        )
+        writer_qos.data_writer_protocol.rtps_reliable_writer.max_send_window_size = (
+            50
+        )
+
+        # Note: Without specifying a name it defaults to DDS_DEFAULT_FLOW_CONTROLLER_NAME. Do I need to specify the name?
```

Comment:
Yes, you have to specify the name. We need to add a constant with the fixed rate name, but for now use this string.